### PR TITLE
create new package for chain tests

### PR DESCRIPTION
**Pull request by rjwebb**
## Description

This PR creates a new package `chain-tests` which exists to hold tests of the chain implementations. It imports the concrete chain implementations from `chain-{ethereum,solana,substrate}` and runs the same tests on them. This should make it easier to integrate new chains in the future because a test suite will already exist to run against them.

This PR introduces a concept of a "WalletMock", which is an object that can stand in for the web extension/wallet for automated testing.

TO DO:
- [x] Implement mock extension class for ethereum
- [x] Implement mock extension class for solana
- [x] Implement mock extension class for substrate
- [x] Move mock extension classes to the chain packages
- [x] Move tests to interfaces package

Tests:
- [x] Sign and verify a valid session payload
- [x] Error is thrown when verifying a session payload that has the wrong `from` value
- [x] Directly sign and verify a valid action payload
- [x] Sign and verify a valid action payload with delegated signer

## How has this been tested?

- [x] CI tests pass
- [ ] Tested with chat-next (including login, all signers, and exchanging messages)
- [ ] Tested with chat-webpack (including login, all signers, and exchanging messages)
- [ ] Other:

## Does this contain any breaking changes to external interfaces?

- [ ] Hooks
- [ ] Daemon API
- [ ] Command line arguments
- [ ] Contract language

### CODE REVIEW

File: packages/chain-substrate/src/implementation.ts [hunk 1]
```diff
@@ -71,39 +71,43 @@ export class SubstrateChainImplementation implements ChainImplementation<Extensi
...
-		const signature = new Buffer(signatureBytes).toString('hex');
+		const signatureBytes = pair.sign(message)
+		const signature = Buffer.from(signatureBytes).toString("hex")
 
 		const action: Action = {
 			type: "action",
 			payload: payload,
 			session: pair.address,
-			signature,
+			signature: `0x${signature}`,
```
Comment (rjwebb) [inline packages/chain-substrate/src/implementation.ts:110]: Do we want to return this? The verify function only works if the signature is prefixed with `0x`.

Comment (joeltg) [reply packages/chain-substrate/src/implementation.ts:110]: if that's what verify expects then definitely yes

File: packages/interfaces/test/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
+{
+  "exclude": ["lib"],
+  "extends": "../../../tsconfig.base.json",
+  "compilerOptions": {
+    "rootDir": ".",
+    "outDir": "lib"
+  },
+  "references": [
+    { "path": "../." },
+    { "path": "../../interfaces" }
```
Comment (joeltg) [inline packages/interfaces/test/tsconfig.json:10]: this is redundant

**Review by joeltg — APPROVED** [review]
nice!

File: packages/interfaces/test/tsconfig.json (hunk 2 above)
Comment (joeltg) [reply packages/interfaces/test/tsconfig.json:10]: oh also I think we need references to the all the `chain-*` packages in here

**Review by joeltg — CHANGES_REQUESTED** [review]

**Pull request merged**